frontend/amundsen_application/static/js/pages/SearchPage/SearchFilter/InputFilter/index.tsx

```diff
@@ -61,40 +76,112 @@ export class InputFilter extends React.Component<
     } = this.props;
     const newValue = e.target.value.toLowerCase();
 
-    this.setState({ value: newValue });
+    if (newValue.includes(',')) {
+      this.setState({ value: newValue, showFilterOperationToggle: true });
+    } else {
+      this.setState({ value: newValue, showFilterOperationToggle: false });
+    }
+
+    const resourceFilter = filterState[resourceType];
+    const inputFilter = resourceFilter ? resourceFilter[categoryId] : undefined;
+    const prevFilterOperation = inputFilter
+      ? inputFilter.filterOperation
+      : undefined;
+    const newFilter = prevFilterOperation
+      ? { value: newValue || undefined, filterOperation: prevFilterOperation }
+      : { value: newValue || undefined };
+
+    const newFilters = {
+      ...filterState,
+      [resourceType]: {
+        ...filterState[resourceType],
+        [categoryId]: newFilter,
+      },
+    };
+    updateFilterState(newFilters);
+  };
+
+  handleFilterOperationChange = (newOperation) => {
+    const {
+      filterState,
+      resourceType,
+      categoryId,
+      updateFilterState,
+    } = this.props;
+
+    this.setState({ filterOperation: newOperation });
+
```

Comment:
We should trigger an event to track this behavior, so we can get analytics information about its usage.